modules/memory.sh

```diff
@@ -1,55 +1,127 @@
 #!/bin/bash
 
-function memory {
-  if [ "$1" == "total" ] ; then
-    memory_total
-    echo "$((t))";
+function memory_total() {
+  option=$1
+  case $option in 
+    '-g')
+      t_M=$(free -m | grep -i Mem | awk '{printf $2}')
+      t=$(echo "scale=2;$t_M/1024" | bc)
+      ;;
+    '-m')
+      t=$(free -m | grep -i Mem | awk '{printf $2}')
+      ;;
+    *)
+      t=$(free | grep -i Mem | awk '{printf $2}')
+      ;;
+  esac
+}
+
+function memory() {
+
+    if [ "$1" == "total" ] ; then
+    memory_total $2
+    echo "$t";
     exit 0
   fi
 
   if [ "$1" == "used" ] ; then
-    memory_used
-    echo "$((ubc))";
+    memory_used $2
+    echo "$ubc";
     exit 0
   fi
 
   if [ "$1" == "free" ] ; then
-    memory_free
-    echo "$((f))";
+    memory_free $2
+    echo "$f";
     exit 0
   fi
+    
 
-  memory_total
-  memory_used
-  memory_free
-  
-  echo "Your rpi has $((t)) bytes of total memory with $((ubc)) bytes used and $((f)) bytes avalaible"
-}
-
-function memory_total {
-  t=$(free | grep -i Mem | awk '{printf $2}')
+ option=$1
+ case $option in
+    '-g')
+      memory_total '-g'
+      memory_used '-g' 
+      memory_free '-g' 
+      echo "Your rpi has $t gigabytes of total memory with $ubc gigabytes used and $f gigabytes avalaible"
+      ;;
+    '-m')
+      memory_total '-m'
+      memory_used '-m' 
+      memory_free '-m' 
+      echo "Your rpi has $t megabytes of total memory with $ubc megabytes used and $f megabytes avalaible"
+      ;;
+    *)
+      if [ ! -n "$1" ]; then
+        option=''
+      else 
+	if [ ! "$1" == 'total' ] && [ ! "$1" == 'used' ] && [ ! "$1" == 'free' ]; then    
+          echo "Error: Ony '-g' and '-m' argument accepted"
+          exit 1
+	fi
+      fi
+      memory_total "$option"
+      memory_used "$option"
+      memory_free "$option" 
+      echo "Your rpi has $t bytes of total memory with $ubc bytes used and $f bytes avalaible"
+  esac
 }
 
 function memory_used {
-  u=$(free | grep -i Mem | awk '{printf $3}')
-  bc=$(free | grep -i Mem | awk '{printf $6}')
-  ubc=$((u+bc))
+  option=$1
+  case $option in 
+    '-g')
+      u_M=$(free -m | grep -i Mem | awk '{printf $3}')
+      u=$(echo "scale=2;$u_M/1024" | bc)
+      bc_M=$(free -m | grep -i Mem | awk '{printf $6}')
+      bc=$(echo "scale=2;$bc_M/1024" | bc)
+      ubc=u+bc|bc
+      #ubc=$( (u+bc) | bc )
```

Comment:
-g failure:
```
$ ./cli.sh memory
Your rpi has 3999784 bytes of total memory with 887628 bytes used and 3111444 bytes avalaible
$./cli.sh memory -g
Your rpi has 3.81 gigabytes of total memory with  gigabytes used and 2.96 gigabytes avalaible
```
The results are missing for memory used in gigabyte argument.
Cause is `ubc=u+bc|bc` should be instead be `$(echo $u + $bc | bc)`

Reply:
I changed the code as you recommended. It should be working now.